src/core/automount.c

```diff
@@ -1038,18 +1037,6 @@ static int automount_dispatch_io(sd_event_source *s, int fd, uint32_t events, vo
                         goto fail;
                 }
 
-                /* Before we do anything, let's see if somebody is playing games with us? */
-                if (lstat(a->where, &st) < 0) {
-                        log_unit_warning_errno(UNIT(a), errno, "Failed to stat automount point: %m");
-                        goto fail;
-                }
-
-                if (!S_ISDIR(st.st_mode) || st.st_dev == a->dev_id) {
-                        log_unit_info(UNIT(a), "Automount point already unmounted?");
-                        automount_send_ready(a, a->expire_tokens, 0);
-                        break;
-                }
-
```

Comment:
Well since the information returned by statx() might be outdated, there might be some cases where the sanity check will fail to notice the automount point is already unmounted, no ?

In this case the deferred unmount job should be prepared to handle this case already. So I'm not sure to understand why we couldn't always let the deferred job do the check...